app/models/tenant.rb

```diff
@@ -1,3 +1,29 @@
 class Tenant < ApplicationRecord
   validates :external_tenant, :uniqueness => true, :presence => true
+
+  after_initialize :setup_settings, :unless => proc { settings.nil? }
+
+  def add_setting(name, value)
+    settings[name] = value
+    save!
+  end
+
+  def update_setting(name, value)
+    add_setting(name, value)
+  end
+
+  def delete_setting(setting)
+    settings.delete(setting.to_s)
```

Comment:
Same here potentially? Should we validate first?

Reply:
Yes definitely. I'll add validation.